geps/gep-1651.md

```diff
@@ -0,0 +1,218 @@
+
+### Status.Addresses
+
+When 'reachability' is specified in the Gateway's `spec` implementations MUST populate the `status.addresses` with
+an address with the desired reachability semantic.
+
+## Examples
+
+#### 1. Request a GatewayAddress that is only reachable within the same cluster
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+#### 2. Request a GatewayAddress with a specific reachability and address
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+  addresses:
+  - value: 10.0.0.8
+```
+#### 3. Request a GatewayAddress that is publically reachable
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+#### 4. Request a GatewayAddress that is a cloud provider's VPC
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    reachability: some.cloud.dev/vpc
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+## Alternatives
+
+### Introducing new GatewayAddress Types
+
+We could introduce additional `AddressTypes` (ie. `ClusterLocalIPAddress`) but
+this would lead to a combinatorial explosion as new dimensions (ie. IPv6) are
+introduced.
+
+See examples in this comment: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1451246877
+
+### Scope/reachability field on GatewayAddress
+
+This would allow Gateways to have multiple scopes.
+
+See pros/cons here: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1486271913
+
+### Survey of Prior Art
+
+These alternatives are a survey of existing approaches to support cluster
+local Gateways. Most are implementation specific and are not portable.
+
+#### Special annotation/label
+
+Istio let's you specify an annotation `networking.istio.io/service-type` to
+change the underlying Kubernetes Service type to make it a ClusterIP type.
+
+#### Re-use of AddressType Hostname
+
+Istio let's you re-use existing Gateway deployments by setting the address
+type to `Hostname` and the value to the Istio ingress Kubernetes Service. If an
+operator configures the Istio deployment to support cluster local traffic a
+Gateway implementation can select it using the `HostName` attribute.
+
+#### Multiple Gateway Classes
+
+Some implementations support multiple deployments on a single cluster where
+each maps to a GatewayClass. One of these deployments can be configured to serve
+cluster local traffic.
```

Comment:
This option feels like it deserves more attention. I'd also note that not every implementation of Gateway API is deployed in-cluster as a deployment.

Reply:
Closing this conversation as a dupe of the discussion here: https://github.com/kubernetes-sigs/gateway-api/pull/1653#discussion_r1160304457